### PR TITLE
[WIP] Transaction gossiping

### PR DESCRIPTION
**Pull request by SkidanovAlex**
This is WIP on gossiping.

Ultimately code will somehow integrate with the Mempool.

### CODE REVIEW

File: core/mempool/src/tx_exchange.rs [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    client: Arc<Client>,
+    payload_receiver: Receiver<ChainPayload>,
+    gossip_receiver: Receiver<Gossip<ChainPayload>>,
+    gossip_sender: Sender<Gossip<ChainPayload>>
+) {
+    let entries: Arc<RwLock<HashMap<CryptoHash, TxExchangeEntry>>> = Arc::new(RwLock::new(HashMap::new()));
+    let entries1 = entries.clone();
+    let payload_recv_task = payload_receiver
+        .for_each(move |payload| {
+            error!("New payload");
```
Comment (SkidanovAlex) [inline core/mempool/src/tx_exchange.rs]: Disregard all the debug outputs, will be removed.

File: core/mempool/Cargo.toml [hunk 2]
```diff
@@ -10,7 +10,9 @@ serde = "1.0"
 serde_derive = "1.0"
 log = "0.4"
 tokio = "0.1"
+bit-set = "0.4.0"
 
+client = { path = "../../node/client" }
```
Comment (bowenwang1996) [inline core/mempool/Cargo.toml]: This will result in circular dependency

File: core/mempool/src/tx_exchange.rs (hunk 1 above)
Comment (bowenwang1996) [inline core/mempool/src/tx_exchange.rs]: This should be `debug`

File: core/mempool/src/tx_exchange.rs [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+            for transaction in payload.transactions.iter() {
```
Comment (bowenwang1996) [inline core/mempool/src/tx_exchange.rs]: I think you want to take ownership of `transaction` here to avoid cloning

File: core/mempool/Cargo.toml [hunk 4]
```diff
@@ -10,7 +10,9 @@ serde = "1.0"
 serde_derive = "1.0"
 log = "0.4"
 tokio = "0.1"
+bit-set = "0.4.0"
```
Comment (bowenwang1996) [inline core/mempool/Cargo.toml]: Looks like this is not used

**Pull request closed without merging**